sormas-backend/src/main/java/de/symeda/sormas/backend/common/AbstractCoreAdoService.java

```diff
@@ -35,4 +50,63 @@ public boolean isArchived(String uuid) {
 		long count = em.createQuery(cq).getSingleResult();
 		return count > 0;
 	}
+
+	protected <T extends ChangeDateBuilder<T>> T addChangeDates(T builder, From<?, ADO> adoPath, boolean includeExtendedChangeDateFilters){
+		return builder.add(adoPath);
+	}
+
+	public Date calculateCaseEndOfProcessingDate(List<String> entityuuids) {
+
+		if (entityuuids.isEmpty()) {
+			return null;
+		}
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Date> cq = cb.createQuery(Date.class);
+		Root<ADO> from = cq.from(getElementClass());
+
+		Expression agregatedChangeDateExpression = addChangeDates(new AgregatedChangeDateExpressionBuilder(cb), from, true).build();
+		cq.select(cb.max(agregatedChangeDateExpression));
+		cq.where(from.get(ADO.UUID).in(entityuuids));
+
+		return em.createQuery(cq).getSingleResult();
+	}
+
+	@TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
+	public void updateArchivedCoreEntities(List<String> entityUuids, Date endOfProcessingDate) {
```

Comment:
Naming doesn't seem right to me, this method is about archiving these entities, right? "updateArchived" sounds like it updates entities that are already archived. Same for the method below. Why not also simply call this `archive` and `dearchive`?

Reply:
renamed